app/src/main/java/com/karumi/ui/presenter/SuperHeroesPresenter.kt

```diff
@@ -1,22 +1,42 @@
 package com.karumi.ui.presenter
 
+import co.metalab.asyncawait.async
 import com.karumi.ui.LifecycleSubscriber
+import com.karumi.ui.domain.Result.Left
+import com.karumi.ui.domain.Result.Right
+import com.karumi.ui.domain.model.SuperHero
+import com.karumi.ui.domain.usecase.GetSuperHeroes
 import java.lang.ref.WeakReference
 
-class SuperHeroesPresenter(view: View) : LifecycleSubscriber {
+class SuperHeroesPresenter(view: View, getSuperHeroes: GetSuperHeroes) : LifecycleSubscriber {
     private val viewWeak: WeakReference<View> = WeakReference(view)
+    private val getSuperHeroes: GetSuperHeroes = getSuperHeroes
 
     private fun view() = viewWeak.get()
 
     override fun initialize() {
-        view()?.hideLoading()
     }
 
     override fun update() {
+        refreshSuperHeroes()
+    }
 
+    private fun refreshSuperHeroes() {
+        async {
+            val result = await { getSuperHeroes.get() }
+            view()?.hideLoading()
+            when (result) {
+                is Right -> view()?.showSuperHeroes(result.value)
```

Comment:
You are not implementing one of the two main kata behaviors. The empty case implementation.

Reply:
because is out of the scope of the PR. I will add in a next iteration.